integration-cli/docker_cli_images_test.go

```diff
@@ -197,3 +196,42 @@ func (s *DockerSuite) TestImagesEnsureDanglingImageOnlyListedOnce(c *check.C) {
 		c.Fatalf("expected 1 dangling image, got %d: %s", a, out)
 	}
 }
+
+func (s *DockerSuite) TestImagesEnsureOnlyHeadsImagesShown(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	dockerfile := `
+        FROM scratch
+        MAINTAINER docker
+        ENV foo bar`
+
+	head, out, err := buildImageWithOut("scratch-image", dockerfile, false)
+	c.Assert(err, check.IsNil)
+
+	split := strings.Split(out, "\n")
+	intermediate := strings.TrimSpace(split[5][7:])
```

Comment:
updated with comment
need intermediate img id, i don't know any other way to get one here